noxfiles/ci_nox.py

```diff
@@ -0,0 +1,163 @@
+"""Contains the nox sessions used during CI checks."""
+import nox
+from constants_nox import CI_ARGS, COMPOSE_SERVICE_NAME, RUN_NO_DEPS, START_APP
+from docker_nox import build
+from run_infrastructure import run_infrastructure
+
+
+@nox.session()
+def ci_suite(session: nox.Session) -> None:
+    """
+    Runs all of the CI checks, except for 'pytest_external'.
+
+    Excludes external tests so that no additional secrets/tooling are required.
+    """
+    # Use "notify" instead of direct calls here to provide better user feedback
+    session.notify("teardown")
+    session.notify("build", ["test"])
+    session.notify("black")
+    session.notify("isort")
+    session.notify("xenon")
+    session.notify("mypy")
+    session.notify("pylint")
+    session.notify("check_install")
+    session.notify("pytest_unit")
+    session.notify("pytest_integration")
+    session.notify("teardown")
+
+
+# Static Checks
+@nox.session()
+def black(session: nox.Session) -> None:
+    """Run the 'black' style linter."""
+    command = (
+        *RUN_NO_DEPS,
+        "black",
+        "--check",
+        "src",
+        "tests",
+        "noxfiles",
+    )
+    session.run(*command, external=True)
+
+
+@nox.session()
+def isort(session: nox.Session) -> None:
+    """Run the 'isort' import linter."""
+    command = (*RUN_NO_DEPS, "isort", "src", "tests", "noxfiles", "--check")
+    session.run(*command, external=True)
+
+
+@nox.session()
+def mypy(session: nox.Session) -> None:
+    """Run the 'mypy' static type checker."""
+    command = (*RUN_NO_DEPS, "mypy")
+    session.run(*command, external=True)
+
+
+@nox.session()
+def pylint(session: nox.Session) -> None:
+    """Run the 'pylint' code linter."""
+    command = (*RUN_NO_DEPS, "pylint", "src", "noxfiles")
+    session.run(*command, external=True)
+
+
+@nox.session()
+def xenon(session: nox.Session) -> None:
+    """Run 'xenon' code complexity monitoring."""
+    command = (
+        "xenon",
+        "noxfiles",
+        "src",
+        "tests",
+        "--max-absolute B",
+        "--max-modules B",
+        "--max-average A",
+        "--ignore 'data, docs'",
+        "--exclude src/fidesops/_version.py",
+    )
+    session.run(*command, external=True)
+
+
+@nox.session()
+def check_install(session: nox.Session) -> None:
```

Comment:
I've never seen this error before 🤔 
can you try a `nox -s teardown clean` first and see if that helps? Might be some cruft from a previous build

The new build process never overlaps the installations (dev does `pip install -e .` and prod doesn't build on that stage)

Reply:
I got the same error with a clean build. I see what you are saying about dev and prod not overlapping so weird that I'm getting this error.

Reply:
its possible some local build directories are getting loaded into the container, I'll update the dockerignore to match fidesctl more closely since I remember having some problems there as well